static/app/components/nav/utils.tsx

```diff
@@ -1,31 +1,49 @@
-import type {ComponentProps} from 'react';
 import type {LocationDescriptor} from 'history';
 
 import type Feature from 'sentry/components/acl/feature';
+import type {MenuItemProps} from 'sentry/components/dropdownMenu';
 import {isItemActive} from 'sentry/components/sidebar/sidebarItem';
 import {SIDEBAR_NAVIGATION_SOURCE} from 'sentry/components/sidebar/utils';
 import normalizeUrl from 'sentry/utils/url/normalizeUrl';
 import type {useLocation} from 'sentry/utils/useLocation';
 
-/** NavItem is the base class for both SidebarItem and SubmenuItem  */
+/**
+ ** NavItem is the base class for both SidebarItem and SubmenuItem
+ */
 interface NavItem {
-  /** User-facing item label, surfaced in the UI. Should be translated! */
+  /**
+   ** User-facing item label, surfaced in the UI. Should be translated!
+   */
   label: string;
-  /** Optionally, props which should be passed to a wrapping `<Feature>` guard */
-  feature?: Omit<ComponentProps<typeof Feature>, 'children'>;
+  /**
+   ** Optionally, props which should be passed to a wrapping `<Feature>` guard
+   */
+  feature?: Omit<React.ComponentProps<typeof Feature>, 'children'>;
 }
 
-/** NavItems are displayed in either `main` and `footer` sections */
+/**
+ ** NavItems are displayed in either `main` and `footer` sections
+ */
 export interface NavItemLayout<Item extends NavSidebarItem | NavSubmenuItem> {
   main: Item[];
   footer?: Item[];
 }
 
-/** SidebarItem is a top-level NavItem which is always displayed in the app sidebar */
+/**
+ ** SidebarItem is a top-level NavItem which is always displayed in the app sidebar
+ */
 export interface NavSidebarItem extends NavItem {
-  /** The icon to render in the sidebar */
-  icon: JSX.Element;
-  /** Optionally, the submenu items to display when this SidebarItem is active */
+  /**
+   ** The icon to render in the sidebar
+   */
+  icon: React.ReactElement;
+  /**
+   ** if te items to display when this SidebarItem is active
+   */
+  menu?: MenuItemProps[];
+  /**
+   ** Optionally, the submenu items to display when this SidebarItem is active
+   */
   submenu?: NavSubmenuItem[] | NavItemLayout<NavSubmenuItem>;
```

Comment:
I wonder if these two are confusing now? Maybe a different name for one of these might make sense

Reply:
Oh good call, totally missed that they are too similar now

Reply:
Fixed in 656b978071c